File: backend/apiserver/models/book.go
//
// Copyright © 2019 Sunho Kim. All rights reserved.
//

package models

import (
	"time"

	"github.com/sunho/webf/wfdb"
)

type Book struct {
	wfdb.DefaultModel `db:"-"`
	ID                string    `db:"id" json:"id" pk:"true"`
	CreatedAt         time.Time `db:"created_at" json:"created_at"`
	UpdatedAt         time.Time `db:"updated_at" json:"updated_at"`
	Name              string    `db:"name" json:"name"`
	Author            string    `db:"author" json:"author"`
	Cover             string    `db:"cover" json:"cover"`
	DownloadLink      string    `db:"download_link" json:"download_link"`
}

File: frontend/ios/App/Models/Book.swift
//
// Copyright © 2019 Sunho Kim. All rights reserved.
//

import Foundation
import RealmSwift

struct Book: Codable {
    let id: String
    let title: String
    let author: String
    let downloadLink: String
    let cover: String?
    let coverType: String?
    let chapters: [String: String]?
}

struct Rate: Codable {
    var rate: Double = 0
}
